### PR TITLE
Atom editor automation maintenance documentation

### PR DESCRIPTION
**Pull request by smurly**
this document explains existing editor automation for contributors to be more equipped to maintain the code base.

### CODE REVIEW

File: testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    pass  # do something to test here starting with imports
+
+if __name__ == "__main__":
+    from editor_python_test_tools.utils import Report
+    Report.start_test(MyTestFunction)
+```
+
+### 2.2 Basic Entity and Components
+
+Components are internally referenced with UUID, but we don't expect anyone to know UUID's and use them so helper methods in editor_entity_utils.py use component names as strings to look up the UUID for you and add components. We utilize atom_constants.py class AtomComponentProperties (which has static methods for each component) to store properties for components including the component name and property paths. By writing AtomComponentProperties.mesh() in your code you'll get back the component name string 'Mesh' which you can use for asking for the addition of a Mesh component like this:
```
Comment (jromnoa) [inline testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md:30]: should we link to `editor_entity_utils.py` and other modules mentioned using the O3DE repo?
i.e. https://github.com/o3de/o3de/blob/development/AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_entity_utils.py 

same for the other modules like `atom_constants.py` = https://github.com/o3de/o3de/blob/development/AutomatedTesting/Gem/PythonTests/Atom/atom_utils/atom_constants.py

File: testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    # my_mesh_component = my_entity.add_component('Mesh')  # this adds a 'Mesh' component to my_entity.
+    # for code maintainability we would use AtomComponentProperties.mesh() which returns 'Mesh' 
+    # that way if the name changes our code only needs to update one place in atom_constants.py
+    my_mesh_component = my_entity.add_component(AtomComponentProperties.mesh())
+
+if __name__ == "__main__":
+    from editor_python_test_tools.utils import Report
+    Report.start_test(MyTestFunction)
+```
+:white_check_mark: Import only the objects you plan to use. In code usage will be cleaner when you `from module imort MyClass`, then you don't need to alias it with `import module as alias` you just call MyClass directly.
```
Comment (jromnoa) [inline testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md:54]: typo: `imort`

File: testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+class TestAutomation(EditorTestSuite):
+
+    class AtomEditorComponents_SkyAtmosphereAdded(EditorBatchedTest):
+        from Atom.tests import hydra_AtomEditorComponents_SkyAtmosphereAdded as test_module
+
+    @pytest.mark.test_case_id("C36525666")
+    class AtomEditorComponents_SSAOAdded(EditorBatchedTest):
+        from Atom.tests import hydra_AtomEditorComponents_SSAOAdded as test_module
+```
+`@pytest.mark.test_case_id("C36525666")` is a deprecated test case ID notation that isn't exposed in any useful way.
```
Comment (jromnoa) [inline testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md:210]: I believe it is exposed on the XML report if you use the CLI arg `--junitxml=C:\path\to\out_report.xml` where that path is where you'll save the XML to. We used to utilize these in past implementations where TestRail was more heavily used, but now we don't.

Might be worth adding that for additional context so users are aware of XML functionality if they wanted to go that route.

File: testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+```
+Use the optional parameter to run specific sets of tests `--tests-regex "AutomatedTesting::Atom_Main_Null"`
+
+Tests are defined in [CMakeLists.txt](https://github.com/o3de/o3de/blob/development/AutomatedTesting/Gem/PythonTests/Atom/CMakeLists.txt)
+
+Note the NAME found in CMakeLists.txt is the what's specified at a CTest command line. A PATH property points to the pytest file that collected editor tests in section 4.0.
+
+TIMEOUT is in seconds. Jenkins will terminate jobs that run longer than 1800 seconds so setting this Ctest timeout to values larger would be terminated by Jenkins. `EditorTestSuite` and `EditorBatchedTest` themselves have a timeout default. Cumulative timeout total for EditorTestSuite should be less than CMakeLists.txt TIMEOUT for a given pytest suite module.
+
+## 6.0 Using Notepad++ to Assmeble Code for New Components
```
Comment (jromnoa) [inline testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md:226]: typo: `Assmeble`

File: testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+You can use Notepad++ search replace once you've figured out a components common name and successfully scripting the creation of that component and dump of the property paths with type as shown in section 2.5.
```
Comment (jromnoa) [inline testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md:228]: nit: we could add a hyperlink back up to section 2.5 here

File: testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+
+- Starting with the list of property paths with type do some simple replace with extended characters to chop off the spew prefix and extra lines.
+- Locate and delete lines for 'Controler' and 'Configuration' since these are actually grouping stubs not properties you would set.
```
Comment (jromnoa) [inline testing-guidance/Automation-Authoring-Maintaining-Editor-Python-Bindings.md:231]: typo: `Controler`

**Review by jromnoa — CHANGES_REQUESTED** [review]

**Review by NeilGitGud — APPROVED** [review]

**Review by jromnoa — APPROVED** [review]

**Pull request merged**